File: src/main/java/com/exhibitions/modernexhibitions/dto/ExhibitionDto.java
package com.exhibitions.modernexhibitions.dto;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.List;

/**
 * ExhibitionDto
 */
@ToString
@Setter
@Getter
public class ExhibitionDto {

    private Integer id;
    private String title;
    private String startDate;
    private String endDate;
    private int startYear;
    private int endYear;
    private int numArtists;
    private int numCatalogueEntries;
    List<LocationDto> locations;

}

File: src/main/frontend/src/app/dtos/ExhibitionDto.ts
import {LocationDto} from "./LocationDto";

export class ExhibitionDto {
  constructor(
    public startDate: string,
    public endDate: string,
    public startYear: number,
    public endYear: number,
    public title: string,
    public numArtists: number,
    public numCatalogueEntries: number,
    public id: number,
    public locations: LocationDto[]
  ){}
}
